File: lib/pacer/pipe/edges_pipe.rb
module Pacer::Pipes

  # This exists because the Blueprints GraphEdgesPipe includes the graph itself in the path
  class EdgesPipe < AbstractPipe
    attr_reader :starts

    def setStarts(starts)
      @starts = starts.first
      self.iter = @starts.getEdges.iterator
    end

    def processNextStart
      iter.next
    end

    def reset
      self.iter = @starts.getEdges.iterator
    end

    private

    attr_accessor :iter
  end
end

File: ext/src/main/java/com/xnlogic/pacer/pipes/EdgesPipe.java
package com.xnlogic.pacer.pipes;

import com.tinkerpop.pipes.AbstractPipe;
import com.tinkerpop.blueprints.Graph;
import com.tinkerpop.blueprints.Edge;
import com.tinkerpop.blueprints.Vertex;
import com.tinkerpop.blueprints.Direction;
import java.lang.Iterable;
import java.util.Iterator;

public class EdgesPipe extends AbstractPipe<Graph, Edge> {
    private Iterator<Edge> iter;
    private Graph starts;
      
    public void setStarts(Iterator<Graph> starts) {
        // TODO: Error checking?
        this.starts = (Graph)starts.next();
        this.iter = this.starts.getEdges().iterator();
    }

    protected Edge processNextStart() {
        return this.iter.next();
    }

    public void reset() {
        this.iter = this.starts.getEdges().iterator();
    }
}
